### PR TITLE
fix: Saving the root route will correctly update types

### LINKED ISSUE #6444: Broken auto generation of types
### Describe the bug

The auto generated types like `PageData` are not updated automatically when editing files in newer svelte-kit versions.

Example: Changing the return type of the `load` function in `+page.server.ts` does not automatically change the `import('./$types').PageData` type in `+page.svelte`

I first thought that it might be an issue with nvim+lsp but I get the same behavior using vscode. 
It seems that this is caused by `svelte-kit sync` not running automatically on write of a file. 

Running `svelte-kit sync` manually generates the correct types. However reopening the editor or restarting the svelte-languageserver is necessary in order to display the changed types correctly in the editor.

I don't remember that I had this issue in versions prior to the big change of the load API.
The behavior is present in a project that I'm currently migrating to `1.0.0-next.406` and in a fresh `1.0.0-next.454` project that I created with `npm init svelte@latest` (see reproduction example)

### Reproduction

**Setup**
- Clone the reproduction repo https://github.com/icalvin102/animated-octo-memory
- Install dependencies `npm i`
- Run dev server `npm run dev` (optional / has no effect for me)

**Check automatic type (re)generation**
- Open `src/routes/+page.server.ts`
- Change property name of return value from `foo` to something else
- Save file
- Open `scr/routes/+page.svelte`
- Check if hover info on `PageData` reflects the above changes
- Repeat a few times

**Check manual type (re)generation**
- Open `src/routes/+page.server.ts`
- Change property name of return value from `foo` to something else
- Save file
- Open `scr/routes/+page.svelte`
- Check if hover info on `PageData` reflects the above changes
- Run `npm run sync`
- Restart svelte language server or restart editor
- Check if hover info on `PageData` reflects the above changes

### Logs

_No response_

### System Info

```shell
Editors:
  NVIM: v0.7.2
  vscodium: 1.70.2.22230-2
  svelte-languageserver: 0.14.33 

System:
  OS: Linux 5.19 Arch Linux
  CPU: (12) x64 Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
  Memory: 12.34 GB / 31.29 GB
  Container: Yes
  Shell: 5.1.16 - /bin/bash
Binaries:
  Node: 16.16.0 - /usr/bin/node
  npm: 8.17.0 - /usr/bin/npm
Browsers:
  Brave Browser: 104.1.42.88
  Firefox: 103.0.2
npmPackages:
  @sveltejs/adapter-auto: next => 1.0.0-next.69
  @sveltejs/kit: next => 1.0.0-next.454
  svelte: ^3.44.0 => 3.49.0
  vite: ^3.1.0-beta.1 => 3.1.0-beta.1
```

### Severity

blocking an upgrade

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by tcc-sejohnson**
`path.dirname` was resolving to '.' for the root route, meaning it wouldn't match the `''` identifier in the manifest. Easy fix.

Closes #6444.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/core/sync/write_types/index.js [hunk 1]
```diff
@@ -72,7 +72,8 @@ export async function write_types(config, manifest_data, file) {
 	}
 
 	const filepath = path.relative(config.kit.files.routes, file);
-	const id = path.dirname(filepath);
+	let id = path.dirname(filepath);
+	id = id === '.' ? '' : id;
```
Comment (Rich-Harris) [inline packages/kit/src/core/sync/write_types/index.js:76]:
```suggestion
```

File: packages/kit/src/core/sync/write_types/index.js [hunk 2]
```diff
@@ -72,7 +72,8 @@ export async function write_types(config, manifest_data, file) {
 	}
 
 	const filepath = path.relative(config.kit.files.routes, file);
```
[the remark this answers is not in the document]
Comment (tcc-sejohnson) [reply packages/kit/src/core/sync/write_types/index.js:74]: here goes nothin

**Comment by tcc-sejohnson** [comment]
@Rich-Harris 

Hang on -- tests may pass, but I didn't add a test to verify this behavior (couldn't figure out how). Do we want to add a test? If no, I need to go manually check that the behavior is working as expected.

**Comment by tcc-sejohnson** [comment]
Can confirm on local repro that the changes we just made do work.

**Pull request merged**